RevenueCatUI/Views/TimelineView.swift

```diff
@@ -0,0 +1,322 @@
+                    .font(.system(size: 11))
+            }
+        } else {
+            HStack(alignment: .top, spacing: 12) {
+                iconWithBackground
+                    .alignmentGuide(.iconAlignment) {
+                        $0[HorizontalAlignment.center]
+                    }
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(configuration.title)
+                        .font(.system(size: 17))
+                        .bold()
+                    Text(configuration.subtitle)
+                        .foregroundStyle(.secondary)
+                        .font(.system(size: 12))
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var iconWithBackground: some View {
+        ZStack {
+            Circle()
+                .fill(configuration.iconBackgroundColor)
+                .overlay(
+                    RoundedRectangle(cornerRadius: iconSize / 2)
+                        .stroke(TimelineView.Colors.link, lineWidth: 1)
+                )
+            Image(systemName: configuration.icon)
+                .resizable()
+                .scaledToFit()
+                .frame(width: iconSize / 2, height: iconSize / 2)
+                .foregroundColor(configuration.iconForegroundColor)
+        }
+        .frame(width: iconSize, height: iconSize)
+    }
+}
+
+#if DEBUG
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+struct Preview: View {
+    
+    @Environment(\.horizontalSizeClass)
+    var horizontalSizeClass
+    
+    var body: some View {
+        if horizontalSizeClass == .regular {
+            TimelineView(stepConfigurations: [
+                TimelineStepView.Configuration(
+                    title: "Today: Free trial for 7 days",
+                    icon: "gift",
+                    iconBackgroundColor: TimelineView.Colors.green,
+                    iconForegroundColor: .black,
+                    subtitle: "Get full access to design and animation tools",
+                    linkColor: TimelineView.Colors.green,
+                    linkPosition: .trailing
+                ),
+                TimelineStepView.Configuration(
+                    title: "Day 5: Reminder",
+                    icon: "bell",
+                    iconForegroundColor: TimelineView.Colors.bell,
+                    subtitle: "We’ll send you a reminder 2 days before your free trial ends.",
+                    linkPosition: .both
+                ),
+                TimelineStepView.Configuration(
+                    title: "Day 7: Trial ends",
+                    icon: "crown.fill",
+                    iconForegroundColor: TimelineView.Colors.linearityOrange,
+                    subtitle: "Your Pro subscription starts, unless you’ve canceled during the trial.",
+                    linkPosition: .leading
+                )
+            ])
+            .frame(width: 326, height: 176)
```

Comment:
I've tried removing frame and wrapping in (as this is how it will be used probably)
```
VStack {
spacer()
TimelineView()
spacer()
}
```
and that didn't work as expected, view took all the available height. Looks like this is because of infinite height rectangle in `linkSegment`.
<img width="505" alt="Screenshot 2024-05-16 at 8 02 31 AM" src="https://github.com/LinearityGmbH/purchases-ios/assets/575303/347ad245-5852-46b5-bac7-02ddda9db8cb">

Reply:
I don't think we should be putting a constant 'height', as that way translations would not be displayed properly.

Reply:
Fixed it. I scratched my head a bit for this one but I managed to do it by replacing the `ZStack` by `.background`